src/script/miniscript.h

```diff
@@ -555,10 +600,110 @@ struct Node {
         return res.has_value();
     }
 
+    internal::Ops CalcOps() const {
+        switch (nodetype) {
+            case Fragment::PK_K: return {0, 0, 0};
+            case Fragment::PK_H: return {3, 0, 0};
+            case Fragment::OLDER: return {1, 0, {}};
```

Comment:
In 85cf12e20201eee4ed23902e1da151f63663cfab "Miniscript: ops limit and stack size computation"

What is the distinction between `{}` and `0`? How com some fragments use 0, and others use `{}`?

It seems like `{}` is being used for fragments that have no dis/satisfaction. However it is used for the hashing fragments and those have a dissatisfaction.

Reply:
While hashes do have dissatisfactions, they are always malleable. The ops counting algorithm only works for non-malleable signing, which will never use these.